torchvision/ops/_utils.py

```diff
@@ -61,3 +61,66 @@ def split_normalization_params(
         else:
             other_params.extend(p for p in module.parameters() if p.requires_grad)
     return norm_params, other_params
+
+
+def set_weight_decay(
+    model: torch.nn.Module,
+    weight_decay: float,
+    norm_weight_decay: Optional[float] = None,
+    norm_classes: Optional[List[type]] = None,
+    custom_keys_weight_decay: Optional[List[Tuple[str, float]]] = None,
+):
+    if not norm_classes:
+        norm_classes = [
+            nn.modules.batchnorm._BatchNorm,
+            nn.LayerNorm,
+            nn.GroupNorm,
+            nn.modules.instancenorm._InstanceNorm,
+            nn.LocalResponseNorm,
+        ]
+    norm_classes = tuple(norm_classes)
+
+    params = {
+        "other": [],
+        "norm": [],
+    }
+    params_weight_decay = {
+        "other": weight_decay,
+        "norm": norm_weight_decay,
+    }
+    custom_keys = []
+    if custom_keys_weight_decay is not None:
+        for key, weight_decay in custom_keys_weight_decay:
+            params[key] = []
+            params_weight_decay[key] = weight_decay
+            custom_keys.append(key)
+
+    def _add_params(module, prefix=""):
+        # We firstly consider norm layers
+        if norm_weight_decay is not None and isinstance(module, norm_classes):
+            params["norm"].extend(p for p in module.parameters() if p.requires_grad)
+        else:
+            for name, p in module.named_parameters(recurse=False):
+                if not p.requires_grad:
+                    continue
+                is_custom_key = False
+                for key in custom_keys:
+                    full_name = f"{prefix}.{name}" if prefix != "" else name
+                    if key in full_name:
```

Comment:
This `in` statement could lead to issues in some corner-case. For example say that a module has multiple convolutions layers named `conv`, `conv2`, ... OR `conv1`, ..., `conv10`. This can lead to the `if` activating in more cases than we want.

Reply:
yes, it maybe a trouble, maybe we should define the type of custom keys (type 0: base_name, type 1: full_name, type2: other case (use `in`)

Reply:
if fact, we can limit the user only use type0 and  type1, so we have the trouble to use `in`

Reply:
How about the following? We should always use equalities instead of `in`. When `key` contains `.` it should be matched against the `full_name` else it needs to be matched against the `name`. We can support only 2 types.

Reply:
it is good. we don't have to define the type.